File: Third_Year/DS/Practicas/P1/Filtros/Cliente.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

#include "Cliente.h"
#include <stdlib.h>
#include <time.h>

Cliente::Cliente(){
    
    string deportes[3] = {"Futbol","Baloncesto","Volleyball"};
    
    float precio_inicial = 5.0;
    int tiempo_alquiler = 20 + rand() % (121 - 20); //Entre 20 y 120 minutos
    int dep = rand() % 3;
    
    int numero_personas = 0;    
    if(dep == 0)
        numero_personas = 11 + rand() % (23 - 11); //Entre 11 y 22 personas
    else if(dep == 1)
        numero_personas = 5 + rand() % (11 - 5); //Entre 5 y 10 personas
    else if(dep == 2)
        numero_personas = 6 + rand() % (13 - 6); //Entre 6 y 12 personas
    
    alquiler.setAlquiler(precio_inicial,deportes[dep],tiempo_alquiler,numero_personas);
}

Alquiler Cliente::getAlquiler(){
    return alquiler;
}

void Cliente::setAlquiler(Alquiler alq){
    alquiler.setAlquiler(alq.getPrecio(),alq.getDeporte(),alq.getTiempo(),alq.getPersonas());
}

File: Third_Year/DS/Practicas/P1/Filtros en Dart/Cliente.dart
import 'Alquiler.dart';
import 'dart:math';

class Cliente{
  Alquiler alquiler = new Alquiler();
  Cliente(){
    var rand = new Random();
    List<String> _deportes = ["Futbol","Baloncesto","Volleyball"];
    double precio_inicial = 5.0;
    int numero_personas = 0;
    int tiempo_alquiler = rand.nextInt(100) + 20;
    int dep = rand.nextInt(3);

    if(dep == 0)
      numero_personas = rand.nextInt(22) + 11;
    else if(dep == 1)
      numero_personas = rand.nextInt(10) + 5;
    else if(dep == 2)
      numero_personas = rand.nextInt(12) + 6;

    alquiler.setAlquiler(precio_inicial, _deportes[dep], tiempo_alquiler, numero_personas);
  }

  Alquiler getAlquiler(){
    return alquiler;
  }

  void setAlquiler(Alquiler alq){
    alquiler.setAlquiler(alq.getPrecio(), alq.getDeporte(), alq.getTiempo(), alq.getPersonas());
  }
}